### PR TITLE
[MI-118] Add Requests resource

### PR DESCRIPTION
**Pull request by miogalang**
Add the requests resource

/cc @jwswj @joseconsador @mmolina @atroche

### References
* JIRA: https://zendesk.atlassian.net/browse/MI-118

### Risks
* Low. We might not be able to access the resource endpoints using the client

### CODE REVIEW

File: tests/Zendesk/API/UnitTests/BasicTest.php [hunk 1]
```diff
@@ -178,4 +179,31 @@ public function assertRequestIs($options, $index = 0)
...
         }
     }
+
+    /**
+     * Test for the endpoint using the given method and endpoint
+     *
+     * @param $classMethod - An array containing [resource name, method name]
+     * @param $endpoint    - An array containing [request method, endpoint path]
+     */
+    protected function assertEndpointCalled($classMethod, $endpoint)
```
Comment (joseconsador) [inline tests/Zendesk/API/UnitTests/BasicTest.php]: how about 
```php
protected function assertEndpointCalled(ResourceAbstract $resource, $method, $endpoint, $method='GET')
```

File: tests/Zendesk/API/UnitTests/BasicTest.php [hunk 2]
```diff
@@ -178,4 +179,31 @@ public function assertRequestIs($options, $index = 0)
...
+    {
+        $this->mockAPIResponses([
+            new Response(200, [], '')
+        ]);
+
+        $queryParams = [
+            'start_time' => 1332034771,
+        ];
+
+        $this->client->$classMethod[0]()->$classMethod[1]($queryParams);
```
Comment (joseconsador) [inline tests/Zendesk/API/UnitTests/BasicTest.php]: and for this one 
```php
call_user_func_array(array($resource, $method), $queryParams)
```

Comment (joseconsador) [inline tests/Zendesk/API/UnitTests/BasicTest.php]: Referring to the method by index seems confusing at first.

**Comment by jwswj** [comment]
:+1:

**Pull request merged**